src/withStyles.jsx

```diff
@@ -214,7 +93,24 @@ export function withStyles(
     if (WrappedComponent.defaultProps) {
       WithStyles.defaultProps = { ...WrappedComponent.defaultProps };
     }
+    WithStyles.contextTypes = contextTypes;
+    // Set statics on the component
+    WithStyles.WrappedComponent = WrappedComponent;
+    WithStyles.displayName = `withStyles(${wrappedComponentName})`;
+    WithStyles = hoistNonReactStatics(WithStyles, WrappedComponent);
+
+    // Make into a pure functional component if requested
+    if (pureComponent) {
+      WithStyles = React.memo(WithStyles);
```

Comment:
Do we need this memo? Will `WithStyles` be created more than once?

Reply:
This is just trying to maintain the `React.PureComponent` option. This makes it so the component doesn't re-render if the the props don't change.